Parkdale-Digital/shopify-webcomponents-test
# Implement product card component

In the pull request description, include:
1. An overview of what you've done.
  
Here's an overview of what I’ve done:

a. Custom Product Card Component: Created a 'ProductCard' web component in 'scripts/main.js' with HTML, CSS, and JavaScript, leveraging a shadow DOM for encapsulated styling. This component pulls product details from attributes and includes an "Add to Cart" button that integrates with Shopify’s AJAX API.

b.Shopify Liquid Integration: Created a 'product-card.liquid' section to render each product within a '<product-card>' custom element. Product data, such as title, price, sale price, and image, are passed via data attributes to the component.

c. Styling in SCSS: Defined styles for the '.product-section', '.product-row', and '.product-card' in 'main.scss', adjusting layout and presentation, including responsive design.

d. Asset Build and Import: Imported 'main.js' and 'main.scss' in 'frontend/entrypoints/main.js' for Vite. After running the build, assets were compiled into the 'assets' directory, with 'main.css', 'main.js', and a manifest for loading files in Shopify.

e. Copying Section to Shopify Theme: Successfully copied 'frontend/web/product-card/sections/product-card.liquid' to the 'sections' folder in the Shopify theme, making the new section available directly in Shopify.

This setup integrates a custom web component in Shopify, using Vite for building and optimizing assets and ensuring that all files are correctly organized for use in the Shopify theme.  

 2. Any considerations or challenges faced.
    
The error is due to rollup-plugin-globlin not finishing its generateBundle hook, causing an early build exit. Replacing it with rollup-plugin-copy resolves this, as it's more stable and compatible with Vite for file copying tasks. After updating, clear the cache with:

rm -rf node_modules
npm install

This should stabilize the build process.

 3. Instructions on how to test the component if necessary.
 
Instructions on How to Test the Component

1. Install Dependencies
   npm install

2. Start the Development Server
     npm run dev
 
3. In a separate terminal, start the Shopify theme development server:
     npm run dev-shop

4. Preview and Customize in Shopify
   - Once the server is running, you’ll see a preview link
        http://127.0.0.1:35913/
   - Click Customize your theme at the theme editor.

5. Add the Section to Your Theme
   - In the Shopify theme editor, navigate to the template where you want to add the new section (e.g., home page, product pages).
   - Add the custom section to the template and add blocks to select products to display in this section.
![image](https://github.com/user-attachments/assets/7b856543-ccd7-4c3b-8b41-7204afb4b5cc)


6. Preview the Storefront
   - After saving changes in the theme editor, preview the storefront to see the custom product web component rendered.

This should give you a clear path to test and view the component on your Shopify storefront.
![image](https://github.com/user-attachments/assets/ad57bbf4-082e-430b-9c0e-d073d45cf474)
